src/common/Datastore.ts

```diff
@@ -133,24 +133,57 @@ export default class Datastore {
   private static setupReplicationRequestSchema () {
     const replicationRequestSchema = new Schema<ReplicationRequest>({
       datasetId: Schema.Types.String,
-      minReplicas: Schema.Types.Number,
-      client: Schema.Types.String,
+      maxReplicas: Schema.Types.Number,
       criteria: Schema.Types.String,
-      status: Schema.Types.String
+      client: Schema.Types.String,
+      urlPrefix: Schema.Types.String,
+      maxPrice: Schema.Types.Number,
+      maxNumberOfDeals: Schema.Types.Number,
+      isVerfied: Schema.Types.Boolean,
+      duration: Schema.Types.Number,
+      isOffline: Schema.Types.Boolean,
+      status: Schema.Types.String,
+      errorMessage: Schema.Types.String
+    }, {
+      timestamps: true
     });
     Datastore.ReplicationRequestModel = mongoose.model<ReplicationRequest>('ReplicationRequest', replicationRequestSchema);
   }
 
   private static setupDealStateSchema () {
     const dealStateSchema = new Schema<DealState>({
-      client: Schema.Types.String,
-      provider: Schema.Types.String,
-      pieceCid: Schema.Types.String,
-      dealId: Schema.Types.Number,
+      client: {
+        type: Schema.Types.String,
+        index: true
+      },
+      provider: {
+        type: Schema.Types.String,
+        index: true
+      },
+      dealCid: Schema.Types.String,
+      dataCid: Schema.Types.String,
+      pieceCid: {
+        type: Schema.Types.String,
+        index: true
+      },
       expiration: Schema.Types.Number,
-      state: Schema.Types.String
+      duration: Schema.Types.Number,
+      price: Schema.Types.Number,
+      verified: Schema.Types.Boolean,
+      state: {
+        type: Schema.Types.String,
+        index: true
+      },
+      replicationRequestId: Schema.Types.String,
+      datasetId: Schema.Types.String,
+      dealId: {
+        type: Schema.Types.Number,
+        index: true
```

Comment:
do you want to create field level indexes or composite indexes?
https://mongoosejs.com/docs/guide.html#indexes